### ISSUE TITLE
UncommentedEmptyConstructor FP when  annotated constructor with @Autowired 

### ISSUE THREAD

**Issue by LynnBroe** [pmd/pmd#4141]
**Affects PMD Version:6.50.0** 

**Rule:UncommentedEmptyConstructor**

<https://pmd.github.io/latest/pmd_rules_java_documentation.html#uncommentedemptyconstructor>

**Description:**

Hi, I think PMD should not report a warning at line 5 when  the constructor is  annotated  with @Autowired, this rule should consider annotation `@Autowired`. Besides, I find a previous [issue](https://github.com/pmd/pmd/issues/357) is related to this problem, and it has been fixed. Thanks.

**Code Sample demonstrating the issue:**

```
import org.springframework.beans.factory.annotation.Autowired;

public class Bug {
    @Autowired
    public Bug() {
    }
}
```

**Expected outcome:**

PMD reports a violation at line 5, but that's wrong. That's a false positive.

**Running PMD through:** *CLI*
